atef/check.py

```diff
@@ -87,12 +87,14 @@ def get(self) -> PrimitiveType:
 
     def compare(self, value: PrimitiveType) -> bool:
         """Compare the provided value with this one, using tolerance settings."""
-        if self.rtol is not None or self.atol is not None:
+        if ((self.rtol is not None or self.atol is not None)
+                and isinstance(value, (int, float))):
```

Comment:
Maybe a `not isinstance(value, (str, bool))` would be better, considering how `int`/`float`-like values tend to be more common.

Reply:
I guess if we assume only PrimitiveType (str, int, float, bool) will be submitted, this is equivalent.  Though is this really a significant time save?

Reply:
> if we assume only PrimitiveType (str, int, float, bool) will be submitted

I may be totally wrong here but I think `PrimitiveType` is more of a _hope_ than perhaps a reality. My recollection of my code here isn't as good as I'd hope, so let me ask: could `np.ndarray` leak in here if we don't have a reduction method set? Could values directly from the caproto backend in ophyd get in here unfiltered? If so, we won't necessarily be limited to `PrimitiveType` - we may have those numpy types, arrays, etc.

Reply:
It looks like we have a default reduction method (average) everywhere, and our data collection would methods would raise if the method were missing.

It was also my understanding that we don't really enforce PrimitiveType inside the code, but trying to deserialize a `Value` that holds a non-primitive type will fail. I think it's fair to say we don't support non-primitive values yet ([issue](https://github.com/pcdshub/atef/issues/160))